sources/packages/backend/apps/api/src/route-controllers/announcements/_tests_/e2e/announcement.institutions.controller.getAnnouncements.e2e-spec.ts

```diff
@@ -0,0 +1,127 @@
+import { HttpStatus, INestApplication } from "@nestjs/common";
+import * as request from "supertest";
+import { Repository } from "typeorm";
+import { Announcement } from "@sims/sims-db";
+import { addDays } from "@sims/utilities";
+import {
+  BEARER_AUTH_TYPE,
+  createTestingAppModule,
+  getInstitutionToken,
+  InstitutionTokenTypes,
+} from "../../../../testHelpers";
+
+describe("AnnouncementInstitutionsController(e2e)-getAnnouncements", () => {
+  let app: INestApplication;
+  let announcementsRepo: Repository<Announcement>;
+  const endpoint = `/institutions/announcements?target=institution-dashboard`;
+
+  beforeAll(async () => {
+    const { nestApplication, dataSource } = await createTestingAppModule();
+    app = nestApplication;
+    announcementsRepo = dataSource.getRepository(Announcement);
+  });
+
+  it("Should return a current announcement.", async () => {
+    // Arrange
+    const now = new Date();
+    const before = addDays(-10, now);
+    const future = addDays(30, now);
+
+    const announcement = new Announcement();
+    announcement.message = "test announcement";
+    announcement.messageTitle = "test title";
+    announcement.target = ["student-dashboard", "institution-dashboard"];
+    announcement.startDate = before;
+    announcement.endDate = future;
+    await announcementsRepo.save(announcement);
+    const institutionUserToken = await getInstitutionToken(
+      InstitutionTokenTypes.CollegeFUser,
+    );
```

Comment:
Please create a factory similar to `createFakeNote` to create fake DB objects.

![image](https://github.com/user-attachments/assets/d2be3ab5-13b7-4761-855e-7a96ba1c37d6)

Reply:
added